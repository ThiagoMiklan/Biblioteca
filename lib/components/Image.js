"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _type_validations = require("../tools/type_validations");

var props_obj = {
  definition: 'string',
  image_definition: 'string',
  src: 'string'
};

var Image = function Image(props) {
  (0, _type_validations.validate)(props, props_obj, "Image");
  var definition = props.definition == undefined ? "image" : "image " + props.definition;
  return /*#__PURE__*/_react["default"].createElement("figure", {
    className: definition,
    onClick: props.onClick
  }, /*#__PURE__*/_react["default"].createElement("img", {
    className: props.image_definition,
    src: props.src
  }));
};

var _default = Image;
exports["default"] = _default;