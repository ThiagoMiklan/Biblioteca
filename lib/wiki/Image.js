"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _react = _interopRequireWildcard(require("react"));

var _reactDom = _interopRequireDefault(require("react-dom"));

require("bulma/css/bulma.css");

var _Image = _interopRequireDefault(require("./components/Image"));

_reactDom.default.render( /*#__PURE__*/_react.default.createElement("div", {
  className: "container is-fluid"
}, /*#__PURE__*/_react.default.createElement(_Image.default, {
  src: "https://bulma.io/images/placeholders/128x128.png",
  definition: "is-16x16"
}, " "), /*#__PURE__*/_react.default.createElement(_Image.default, {
  src: "https://bulma.io/images/placeholders/128x128.png",
  definition: "is-24x24"
}, " "), /*#__PURE__*/_react.default.createElement(_Image.default, {
  src: "https://bulma.io/images/placeholders/128x128.png",
  definition: "is-32x32"
}, " "), /*#__PURE__*/_react.default.createElement(_Image.default, {
  src: "https://bulma.io/images/placeholders/128x128.png",
  definition: "is-48x48"
}, " "), /*#__PURE__*/_react.default.createElement(_Image.default, {
  src: "https://bulma.io/images/placeholders/128x128.png",
  definition: "is-64x64"
}, " "), /*#__PURE__*/_react.default.createElement(_Image.default, {
  src: "https://bulma.io/images/placeholders/128x128.png",
  definition: "is-96x96"
}, " "), /*#__PURE__*/_react.default.createElement(_Image.default, {
  src: "https://bulma.io/images/placeholders/128x128.png",
  definition: "is-128x128"
}, " "), /*#__PURE__*/_react.default.createElement(_Image.default, {
  src: "https://bulma.io/images/placeholders/128x128.png",
  definition: "is-128x128",
  image_definition: "is-rounded"
}, " "), /*#__PURE__*/_react.default.createElement(_Image.default, {
  src: "https://bulma.io/images/placeholders/128x128.png",
  definition: "is-128x128",
  image_definition: "is-rounded",
  onClick: function onClick() {
    return alert("Clicked");
  }
})), document.getElementById("root"));