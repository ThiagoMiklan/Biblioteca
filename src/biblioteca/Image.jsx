// @flow 
import React from 'react';
import PropTypes from 'prop-types';
type Props ={
    definition?:string,
    src?:string
}

const Image = (props:Props)=>{
    return (<figure className={props.definition} >
            <img src={props.src}/>
        </figure>);
}

Image.propTypes = {
    definition: PropTypes.string.isRequired,
    src: PropTypes.string.isRequired
}

export default Image;