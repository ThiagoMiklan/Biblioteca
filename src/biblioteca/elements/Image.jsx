import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import getClassName from '../../tools/getClassName';

class Image extends React.Component {

    render() {
        return (
        <figure className={classnames(getClassName(this.props.definition, "Image"))} >
            <img src={this.props.src}/>
        </figure>);
    }

}

Image.propTypes = {
    definition: PropTypes.string,
    value: PropTypes.string,
    src: PropTypes.string
}

export default Image;