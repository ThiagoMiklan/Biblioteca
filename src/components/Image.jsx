// @flow 
import React from 'react';
import PropTypes from 'prop-types';

type Props ={
    definition?:string,
    src?:string
}

const Image = (props:Props)=>{
    return (<figure className={"image "+props.definition} >
            <img className={props.image_definition} src={props.src}/>
        </figure>);
}

Image.propTypes = {
    definition: PropTypes.string.isRequired,
    image_definition: PropTypes.string,
    src: PropTypes.string.isRequired
}

export default Image;