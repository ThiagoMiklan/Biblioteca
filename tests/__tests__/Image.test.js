import React from 'react';
import renderer from 'react-test-renderer';
import Image from '../../src/components/Image';

const tags = [
  {value:'One', definition:'is-primary'},
  {value:'Two', definition:'is-success'},
  {value:"Three",definition:'is-info'},
  {value:"Four",definition:'is-link'},
  {value:"Five",definition:'is-warning'},
  {value:"Six", definition:'vdanger'}
];
const tags_addons =[
  {value:'Package'},
  {value:'Bulma', definition:'is-primary'},
]


test('images normal',()=>{
    const component = renderer.create(
        <>
        <Image src="https://bulma.io/images/placeholders/128x128.png" definition="is-16x16"> </Image>
        <Image src="https://bulma.io/images/placeholders/128x128.png" definition="is-24x24"> </Image>
        <Image src="https://bulma.io/images/placeholders/128x128.png" definition="is-32x32"> </Image>
        <Image src="https://bulma.io/images/placeholders/128x128.png" definition="is-48x48"> </Image>
        <Image src="https://bulma.io/images/placeholders/128x128.png" definition="is-64x64"> </Image>
        <Image src="https://bulma.io/images/placeholders/128x128.png" definition="is-96x96"> </Image>
        <Image src="https://bulma.io/images/placeholders/128x128.png" definition="is-128x128"> </Image>  
        </>
  ).toJSON();
    expect(component).toMatchSnapshot();
});


test('image rounded',()=>{
    const component = renderer.create(
        <>
       <Image src="https://bulma.io/images/placeholders/128x128.png" definition="is-128x128" image_definition="is-rounded"> </Image> 
        </>
  ).toJSON();
    expect(component).toMatchSnapshot();
});

test('image event',()=>{
    const component = renderer.create(
       <Image src="https://bulma.io/images/placeholders/128x128.png"
              definition="is-128x128" image_definition="is-rounded" 
              onClick={()=> alert("Clicked")}/> 
  ).toJSON();
    expect(component).toMatchSnapshot();
});



