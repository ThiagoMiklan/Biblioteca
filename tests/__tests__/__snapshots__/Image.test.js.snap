// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image event 1`] = `
<figure
  className="image is-128x128"
  onClick={[Function]}
>
  <img
    className="is-rounded"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`image rounded 1`] = `
<figure
  className="image is-128x128"
>
  <img
    className="is-rounded"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`images normal 1`] = `
Array [
  <figure
    className="image is-16x16"
  >
    <img
      src="https://bulma.io/images/placeholders/128x128.png"
    />
  </figure>,
  <figure
    className="image is-24x24"
  >
    <img
      src="https://bulma.io/images/placeholders/128x128.png"
    />
  </figure>,
  <figure
    className="image is-32x32"
  >
    <img
      src="https://bulma.io/images/placeholders/128x128.png"
    />
  </figure>,
  <figure
    className="image is-48x48"
  >
    <img
      src="https://bulma.io/images/placeholders/128x128.png"
    />
  </figure>,
  <figure
    className="image is-64x64"
  >
    <img
      src="https://bulma.io/images/placeholders/128x128.png"
    />
  </figure>,
  <figure
    className="image is-96x96"
  >
    <img
      src="https://bulma.io/images/placeholders/128x128.png"
    />
  </figure>,
  <figure
    className="image is-128x128"
  >
    <img
      src="https://bulma.io/images/placeholders/128x128.png"
    />
  </figure>,
]
`;
